### PR TITLE
Allow staking accounts beyond 100'

### PR DESCRIPTION
**Pull request by gabrielKerekes**
fixes https://github.com/vacuumlabs/trezor-firmware/issues/43

We've also added explicit warnings when using accounts over 100.

## Get base address
- If `path` account is larger than 100 an _unknown path_ warning is shown. 
- If `staking_path` account is the same as `path` account (thus also larger than 100) no additional warning is shown, since the user has already been warned about the large account. 
- If `staking_path` account is different from `path` account a warning is always shown regardless of account value.
## Sign transaction
A warning is shown for any non-standard path in:
- input paths
- change output paths (separate warning for payment and staking path)
- certificate paths
- stake pool owner staking key path

path_warnings

### CODE REVIEW

File: core/src/apps/cardano/layout.py [hunk 1]
```diff
@@ -54,6 +54,11 @@
     CardanoCertificateType.STAKE_POOL_REGISTRATION: "Stakepool registration",
 }
 
+WARNING_TITLE_CHANGE_OUTPUT_PATH = "Change output path"
```
Comment (refi93) [inline core/src/apps/cardano/layout.py:57]: Given that we aren't doing it anywhere else (AFAIK) I wouldn't introduce those warning title constants, especially if the given string is used once anyway

**Comment by refi93** [comment]
> I am not sure whether it is OK to use SCHEMA_STAKING_ANY_ACCOUNT in `_validate_pool_parameters.

- as long as we warn about it it's fine. Moreover the stake pool registration certificate can be updated even if the owner account was messed up somehow

**Review by refi93 — APPROVED** [review]
one minor remark, otherwise LGTM

File: core/src/apps/cardano/layout.py (hunk 1 above)
Comment (gabrielKerekes) [reply core/src/apps/cardano/layout.py:57]: True. I tried various structuring of those functions so the constants come from there. They aren't really needed in this final form.

**Comment by refi93** [comment]
closing in favor of https://github.com/trezor/trezor-firmware/pull/1423

**Pull request closed without merging**